File: 2580-circular-sentence/2580-circular-sentence.cpp
class Solution {
public:
    bool isCircularSentence(string sentence) {
        vector<string> words;
        stringstream ss(sentence);
        string word;
        while(ss >> word) {
            words.push_back(word);
        }
        int n = words.size();
        int nsz = words[n-1].size();
        if(words[0][0] != words[n-1][nsz - 1]) return false;
        for(int i = 1; i < n; i++) {
            if(words[i][0] != words[i-1][words[i-1].size() - 1]) return false;
        }
        return true;
    }
};

File: 2580-circular-sentence/2580-circular-sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        words = sentence.split()
        if words[0][0] != words[-1][-1]: 
            return False
        for i in range(1, len(words)):
            if words[i][0] != words[i-1][-1]:
                return False
        return True